router/core/router.go

```diff
@@ -799,6 +799,12 @@ func (r *Router) newServer(ctx context.Context, routerConfig *nodev1.RouterConfi
 		MaxOperationSizeInBytes: int64(r.routerTrafficConfig.MaxRequestBodyBytes),
 		PersistentOpClient:      r.cdnPersistentOpClient,
 	})
+	// Pre-hash all data source IDs to avoid races
```

Comment:
Could you add a few more comments why we need to add this here? For me it's also a TODO to do it in a different place at the engine side.

Reply:
done